crates/consensus/src/crypto.rs

```diff
@@ -73,6 +175,13 @@ pub mod secp256k1 {
         sig[32..64].copy_from_slice(&signature.s().to_be_bytes::<32>());
         sig[64] = signature.v() as u8;
 
+        // Try dynamic backend first when crypto-backend feature is enabled
+        #[cfg(feature = "crypto-backend")]
+        if let Some(provider) = super::backend::try_get_provider() {
+            return provider.recover_signer_unchecked(&sig, &hash.0);
+        }
```

Comment:
If crypto-baackend feature-flag is set, then this will call whatever backend is set here.

To install your own backend, you can call `install_default_provider` in your own codebase. `install_default_provider` can only be called once, calling it more than once will return an error